### PR TITLE
internal: wrap_return_type_in_result works on the HIR

### PR DESCRIPTION
**Pull request by Veykril**
bors r+

### CODE REVIEW

**Comment by Veykril** [comment]
bors retry

**Comment by Veykril** [comment]
bors retry

File: crates/ide_assists/src/handlers/unmerge_use.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+use ast::make;
```
Comment (matklad) [inline crates/ide_assists/src/handlers/unmerge_use.rs:1]: auto-import haven’t figured out that we use ast::self already?

File: crates/ide_assists/src/handlers/wrap_return_type_in_result.rs [hunk 2]
```diff
@@ -98,10 +105,17 @@ mod tests {
 
     use super::*;
 
-    #[test]
-    fn wrap_return_type_in_result_simple() {
+    fn check(ra_fixture_before: &str, ra_fixture_after: &str) {
         check_assist(
             wrap_return_type_in_result,
+            &format!("//- minicore: result\n{}", ra_fixture_before.trim_start()),
```
Comment (matklad) [inline crates/ide_assists/src/handlers/wrap_return_type_in_result.rs:111]: I am not a huge fan of hiding meta lines in the check function. Repeating `//- minicore:result` everywhere is formal duplication, but semantically, it is not. You fundamentally need to signal, in each test, whether it uses minicore or not.

By extracting this into a function, the decision becomes hidden.

File: crates/ide_assists/src/handlers/unmerge_use.rs (hunk 1 above)
Comment (Veykril) [reply crates/ide_assists/src/handlers/unmerge_use.rs:1]: Huh that is odd. I think I used the qualify path assist here, so maybe tha has still problems with this.

File: crates/ide_assists/src/handlers/wrap_return_type_in_result.rs (hunk 2 above)
Comment (Veykril) [reply crates/ide_assists/src/handlers/wrap_return_type_in_result.rs:111]: Fair point, I'll correct that then.

**Pull request merged**